src/lib/journey/_utilities/map-callback.utilities.ts

```diff
@@ -10,13 +10,14 @@ import Name from '$journey/callbacks/username/name.svelte';
 import Password from '$journey/callbacks/password/password.svelte';
 import StringAttributeInput from '$journey/callbacks/string-attribute/string-attribute-input.svelte';
 import TermsConditions from '$journey/callbacks/terms-and-conditions/terms-conditions.svelte';
+import Unknown from '$journey/callbacks/unknown/unknown.svelte';
 
 /**
  * Iterate through callbacks received from AM and map the callback to the
  * appropriate callback component, pushing that component
  * the StepComponent's array.
  */
-export function mapCallbackToComponent(cb: any) {
+export function mapCallbackToComponent(cb: FRCallback) {
```

Comment:
I read your comment on [[src/lib/_utilities/i18n.utilities.ts](https://github.com/cerebrl/forgerock-web-login-framework/pull/18/files/0ceb5f624300f6f2b82824def819da53f86668c3#diff-d521444e98934562466afd9f553b7b765b61b1d56f36cf33cdf5e8b69739213b)](https://github.com/cerebrl/forgerock-web-login-framework/pull/18/files/0ceb5f624300f6f2b82824def819da53f86668c3#diff-9a05e063ad2fef3b804a33373dbd21f71a799916be581e8289aee521b218bf7d) about there not being enough overlap between callbacks to actually use FRCallback, and instead you used `any`. I can see here you've swapped `any` for `FRCallback` and I am wondering do we have an issue with the types in the SDK, if not all of them are comparable / are subclasses of the FRCallback class? Maybe I'm a bit confused but it would be good to know the issue there

Reply:
Yeah, the way the SDK is typed, this gets a bit hairy when assignment is done in templates. Like the call site of this very function in the stage component. The issue is I can't assign types in the template portion of Svelte (the HTML essentially), so I have to make adjustments higher up in the code flow.

I'm not entirely sure what our long term plan is concerning this, but considering the downstream components are properly typed, I'm not super concerned about this. I doubt properly typing this would lead to an equivalent value add to the actual code itself.

Switching this from `any` to `FRCallback` didn't produce any side-effects, and it removes another `any` from the codebase. We now just have the one from the portion of code you linked to.